client/my-sites/upgrades/checkout-thank-you/header/index.jsx

```diff
@@ -42,13 +43,27 @@ const CheckoutThankYouHeader = React.createClass( {
 		}
 
 		if ( this.props.primaryPurchase ) {
-			return this.translate(
-				"You will receive an email confirmation shortly for your purchase of %(productName)s. What's next?", {
+			if ( this.props.isFreeTrial ) {
```

Comment:
`primaryPurchase` has a free trial property, so we can use `isFreeTrial( this.props.primaryPurchase )` here and obviate that prop entirely.